static/app/views/insights/common/utils/useStarredTransactions.tsx

```diff
@@ -0,0 +1,41 @@
+import {useCallback} from 'react';
+
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+
+export function useStarredTransactions() {
+  const organization = useOrganization();
+  const api = useApi();
+
+  const starTransaction = useCallback(
+    (segmentName: string, projectId: string) =>
+      api.requestPromise(
+        `/organizations/${organization.slug}/insights/starred-segments/`,
+        {
+          method: 'POST',
+          data: {
+            project_id: projectId,
+            segment_name: segmentName,
+          },
+        }
+      ),
+    [api, organization.slug]
+  );
+
+  const unstarTransaction = useCallback(
+    (segmentName: string, projectId: string) =>
+      api.requestPromise(
+        `/organizations/${organization.slug}/insights/starred-segments/`,
+        {
+          method: 'DELETE',
+          data: {
+            project_id: projectId,
+            segment_name: segmentName,
+          },
+        }
+      ),
+    [api, organization.slug]
+  );
```

Comment:
I'm honestly not familiar with the API, but is it possible to use `useMutation` here? Seems like it's pretty popular in the codebase, maybe it has some noticeable advantage?

Reply:
Interesting suggestion! I'll take a look at it